common/src/main/java/com/datastrato/gravitino/json/JsonUtils.java

```diff
@@ -1096,4 +1103,49 @@ public PartitionDTO deserialize(JsonParser p, DeserializationContext ctxt) throw
       }
     }
   }
+
+  public static class IndexSerializer extends JsonSerializer<Index> {
+    @Override
+    public void serialize(Index value, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeStartObject();
+      gen.writeStringField(INDEX_TYPE, value.type().name().toUpperCase(Locale.ROOT));
+      if (null != value.name()) {
+        gen.writeStringField(INDEX_NAME, value.name());
+      }
+      gen.writeArrayFieldStart(INDEX_FIELD_NAMES);
+      for (String[] fieldName : value.fieldNames()) {
+        gen.writeArray(fieldName, 0, fieldName.length);
+      }
+      gen.writeEndArray();
+      gen.writeEndObject();
+    }
+  }
+
+  public static class IndexDeserializer extends JsonDeserializer<Index> {
+    @Override
+    public Index deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode node = p.getCodec().readTree(p);
+      Preconditions.checkArgument(
+          node != null && !node.isNull() && node.isObject(),
+          "Cannot parse index from invalid JSON: %s",
```

Comment:
The error message can be more detail, for example `"Index must be a valid JSON object, but found: %s"`